packages/jellyfish-api-core/__tests__/category/loan/createVault.test.ts

```diff
@@ -0,0 +1,94 @@
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { LoanMasterNodeRegTestContainer } from './loan_container'
+import { UTXO } from '@defichain/jellyfish-api-core/category/oracle'
+
+describe('Loan', () => {
+  const container = new LoanMasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+
+    // NOTE(jingyi2811): default scheme
+    await container.call('createloanscheme', [100, 1, 'default'])
+    await container.generate(1)
+  })
+
+  afterEach(async () => {
+    const result = await container.call('listloanschemes')
+    const data = result.filter((r: { default: boolean }) => !r.default)
+
+    for (let i = 0; i < data.length; i += 1) {
+      await container.call('destroyloanscheme', [data[i].id])
+      await container.generate(1)
+    }
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should create vault', async () => {
+    await container.call('createloanscheme', [200, 2, 'scheme'])
+    await container.generate(1)
+
+    const vaultId = await client.loan.createVault(await container.getNewAddress(), 'scheme')
+
+    // const tx: any = await client.call('getrawtransaction', [vaultId, true], 'bignumber')
+    // console.log(tx.vout[0].scriptPubKey)
+
+    expect(typeof vaultId).toStrictEqual('string')
+    expect(vaultId.length).toStrictEqual(64)
+
+    await container.generate(1)
+  })
+
+  it('should not create vault with invalid address', async () => {
+    await container.call('createloanscheme', [200, 2, 'scheme'])
+    await container.generate(1)
+
+    const promise = client.loan.createVault('INVALID_OWNER_ADDRESS', 'scheme')
+    await expect(promise).rejects.toThrow('Error: Invalid owneraddress address\', code: -5, method: createvault')
+  })
+
+  it('should not create vault with not exists loan scheme id', async () => {
+    const promise = client.loan.createVault(await container.getNewAddress(), 'scheme')
+    await expect(promise).rejects.toThrow('pcApiError: \'Test VaultTx execution failed:\nCannot find existing loan scheme with id scheme\', code: -32600, method: createvault')
+  })
+
```

Comment:
I think these two test should come after line 61. Positive tests first before negatives.

Reply:
So far there is no deleteVault rpc call